src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -9,35 +9,27 @@ namespace System.Threading
 {
     internal partial class PortableThreadPool
     {
-        private class CpuUtilizationReader
+        private struct CpuUtilizationReader
         {
-            private struct ProcessCpuInformation
-            {
-                public long idleTime;
-                public long kernelTime;
-                public long userTime;
-            }
-
-            private ProcessCpuInformation _processCpuInfo = new ProcessCpuInformation();
+            public long _idleTime;
+            public long _kernelTime;
+            public long _userTime;
 
             public int CurrentUtilization
             {
                 get
                 {
                     if (!Interop.Kernel32.GetSystemTimes(out var idleTime, out var kernelTime, out var userTime))
                     {
-                        int error = Marshal.GetLastWin32Error();
-                        var exception = new OutOfMemoryException();
-                        exception.HResult = error;
-                        throw exception;
+                        return 0;
                     }
 
-                    long cpuTotalTime = ((long)userTime - _processCpuInfo.userTime) + ((long)kernelTime - _processCpuInfo.kernelTime);
-                    long cpuBusyTime = cpuTotalTime - ((long)idleTime - _processCpuInfo.idleTime);
+                    long cpuTotalTime = ((long)userTime - _userTime) + ((long)kernelTime - _kernelTime);
+                    long cpuBusyTime = cpuTotalTime - ((long)idleTime - _idleTime);
```

Comment:
Do we have any tests that would fall if we got something wrong here?

Reply:
Currently, this is only used for Mono on Windows that gets a very light testing. If we got this wrong, this would manifest itself as a performance bug in thread pool scaling.

Hopefully, we will find time to switch to the managed threadpool for CoreCLR too. Validating  details like this would be the bulk of the work for that.